cmd/incus/storage_volume.go

```diff
@@ -612,15 +616,29 @@ func (c *cmdStorageVolumeCreate) Run(cmd *cobra.Command, args []string) error {
 
 	client := resource.server
 
+	var volumePut api.StorageVolumePut
+	if !termios.IsTerminal(getStdinFd()) {
+		contents, err := io.ReadAll(os.Stdin)
+		if err != nil {
+			return err
+		}
+
+		err = yaml.UnmarshalStrict(contents, &volumePut)
+		if err != nil {
+			return err
+		}
+	}
+
 	// Parse the input
 	volName, volType := parseVolume("custom", args[1])
 
 	// Create the storage volume entry
-	vol := api.StorageVolumesPost{}
-	vol.Name = volName
-	vol.Type = volType
-	vol.ContentType = c.flagContentType
-	vol.Config = map[string]string{}
+	vol := api.StorageVolumesPost{
+		Name:             volName,
+		Type:             volType,
+		ContentType:      c.flagContentType,
+		StorageVolumePut: volumePut,
+	}
```

Comment:
This causes build failures as you are no longer initializing `Config` to an empty map.